woocommerce/woocommerce-gutenberg-products-block
# Render pay for order checkout block if it is pay for order order

Blocked by https://github.com/woocommerce/woocommerce-blocks/pull/9448

Instead of rendering checkout error, render pay-for-order checkout block if it is a pay-for-order order

Fixes #9459 



### Screenshots



| Before | After |
| ------ | ----- |
|<img width="586" alt="Screen Shot 2023-05-12 at 5 30 28 PM" src="https://github.com/woocommerce/woocommerce-blocks/assets/56378160/654af3cc-2828-4d65-8bb3-de8b9e4ffc7e">|<img width="668" alt="Screen Shot 2023-05-12 at 6 38 17 PM" src="https://github.com/woocommerce/woocommerce-blocks/assets/56378160/49756940-9935-4928-b172-cf4be6fee0a6">
    |

### Testing
1. Create a pay-for-order order
2. Go to WC -> Orders -> find the order you just created
3. Click on the Customer payment page
4. Should see the pay-for-order checkout block
5. Open the same URL in incognito
6. Make sure the page renders without error

#### Automated Tests


### Changelog

> Render pay for order checkout block if it is pay for order order
